prime-dotnet-webapi/Controllers/GisController.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            if (changeModel == null)
+            {
+                ModelState.AddModelError("GisEnrolment", "Profile update model cannot be null.");
+                return BadRequest(ApiResponse.BadRequest(ModelState));
+            }
+
+            var gisEnrolment = await _gisService.GetGisEnrolmentByIdAsync(gisId);
+            if (gisEnrolment == null)
+            {
+                return NotFound(ApiResponse.Message($"Gis Enrolment not found with id {gisId}"));
+            }
+            if (!gisEnrolment.Party.PermissionsRecord().AccessableBy(User))
+            {
+                return Forbid();
+            }
+
+            if (!changeModel.Validate(User))
+            {
+                ModelState.AddModelError("GisEnrolment", "One or more Properties did not match the information on the card.");
+                return BadRequest(ApiResponse.BadRequest(ModelState));
+            }
+
+            await _gisService.CreateOrUpdateGisEnrolmentAsync(changeModel, User);
+
+            return NoContent();
+        }
+
+        // GET: api/Gis/5
+        /// <summary>
+        /// Gets a specific Gis Enrolment.
+        /// </summary>
+        /// <param name="gisId"></param>
+        [HttpGet("{gisId}", Name = nameof(GetGisEnrolmentById))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<GisViewModel>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<GisViewModel>> GetGisEnrolmentById(int gisId)
+        {
+            var gisEnrolment = await _gisService.GetGisEnrolmentByIdAsync(gisId);
+            if (gisEnrolment == null)
+            {
+                return NotFound(ApiResponse.Message($"Gis Enrolment not found with id {gisId}"));
+            }
+            if (!gisEnrolment.Party.PermissionsRecord().AccessableBy(User))
+            {
+                return Forbid();
+            }
+
+            return Ok(ApiResponse.Result(gisEnrolment));
+        }
+
+        // GET: api/Gis/5
+        /// <summary>
+        /// Gets a specific Gis Enrolment using the logged in User.
+        /// </summary>
+        [HttpGet("", Name = nameof(GetGisEnrolmentByUserId))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<GisViewModel>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<GisViewModel>> GetGisEnrolmentByUserId()
```

Comment:
This is how we do this kind of thing in our app currently for other types of users (Enrollees, etc), but I would prefer to do away with this pattern (and eventually refactor those other endpoints as well).  I feel this should be an endpoint that gets the enrollment by UserId (api/parties/gis/{userId:guid}) and verifies the UserId, rather that silently pulling the UserId from the token "behind the scenes". It would also allow Admin users to hit this endpoint if we needed to.